tests/integration/test_styles_file_download_button.py

```diff
@@ -0,0 +1,56 @@
+from PyQt5.QtCore import QEvent
+from PyQt5.QtGui import QFont, QPalette
+
+from securedrop_client.resources import load_icon
+
+
+def test_styles(mocker, main_window):
+    wrapper = main_window.main_view.view_layout.itemAt(0).widget()
+    conversation_scrollarea = wrapper.conversation_view.scroll
+    file_widget = conversation_scrollarea.widget().layout().itemAt(0).widget()
+    download_button = file_widget.download_button
+
+    expected_image = load_icon('download_file.svg').pixmap(20, 20).toImage()
+    assert download_button.icon().pixmap(20, 20).toImage() == expected_image
+    assert 'Source Sans Pro' == download_button.font().family()
+    assert QFont.Bold == download_button.font().weight()
+    assert 13 == download_button.font().pixelSize()
+    assert '#2a319d' == download_button.palette().color(QPalette.Foreground).name()
+    # assert 'border: none;' for download_button
+
+    file_widget.eventFilter(download_button, QEvent(QEvent.HoverEnter))
+    expected_image = load_icon('download_file_hover.svg').pixmap(20, 20).toImage()
+    assert download_button.icon().pixmap(20, 20).toImage() == expected_image
+    # assert '#05a6fe' == download_button.palette().color(QPalette.Foreground).name()
```

Comment:
`HoverEnter` is the right event for to test the setting of the icon since that logic is [explicit here](https://github.com/freedomofpress/securedrop-client/blob/master/securedrop_client/gui/widgets.py#L2414-L2417), but it's not obvious to me where the application of the `hover` selector is happening since it's not done in the `eventFilter`, so I'm not sure how to test this properly (i.e. _is_ `HoverEnter` the right event?). 

One thought prior to spending more time on this: it seems like the easiest way to test more dynamic behavior would be to actually move the cursor to hover over the `download_button` (similar to [this](https://github.com/freedomofpress/securedrop-client/blob/master/tests/functional/test_user_icon_click.py#L45) but using `pyautogui.moveTo`) and then assert the color on the widget (and ultimately take screenshots). This doesn't necessarily need to be part of this change, but what do you think @creviera @kushaldas ?

Reply:
I think it makes sense to test longer sequences of client behavior using pyautogui or some other similar tool. Here we just want to check the that a button's hover event or pressed event updates the button's palette color correctly. I'm surprised there isn't an easy way to do this without actually moving the mouse cursor. Just for a little background, here are two things I tried for the login button:

1. Creating a mouse event and triggering it.
   ```py
   event = QMouseEvent(QEvent.MouseButtonPress, login_button.pos(), Qt.LeftButton, Qt.LeftButton, Qt.NoModifier)
   login_button.mousePressEvent(event)
   ```

   Or triggering it this way:
   ```py
   event = QMouseEvent(QEvent.MouseButtonPress, login_button.pos(), Qt.LeftButton, Qt.LeftButton, Qt.NoModifier)
   QCoreApplication.postEvent(main_window, event)
   ```

2.  Using Qt's [QTest unit testing framework](https://doc.qt.io/qt-5/qtest-overview.html) which offers functionality for mouse and keyboard simulation.
    ```py
    QTest.mousePress(login_button, Qt.LeftButton)
    ```

    Or, for the download button scenario a mouse hover event:
    ```py
    QTest.mouseMove(download_button)
    ```


But these simulations don't cause the buttons to update their palette color, even though it clearly works when we run the client. I'm starting to wonder if there is another color property somewhere... At this point I think it would be good to get @kushaldas's eyes on this PR and to move forward without full coverage. I'll create a couple followup issues for addressing the remaining work.